portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -23,37 +24,48 @@
     /**
      * The date the user wrote the message in dd/mm/yyyy format.
      */
-    private String datePosted;
+    private Long datePosted;
 
     /**
      * Constructor to use when receiving input from user.
      */
     protected Comment(String text){
         this.text = text;
-        this.datePosted = convertDateToString(new Date());
+        this.datePosted = System.currentTimeMillis();
     }
 
     /**
      * Constructor to use when receiving input from Datastore.
      */
-    protected Comment(Long id, String text, String datePosted){
+    protected Comment(Long id, String text, Long datePosted){
         this.id = id;
         this.text = text;
         this.datePosted = datePosted;
     }
 
+    private static DateFormat formatter = new SimpleDateFormat("dd/MM/yyyy");
+
     /**
      * Helper function to turn Date object into
      * String using dd/mm/yyyy format.
      * @param date The date object to convert.
      * @return The String representation.
      */
     private static String convertDateToString(Date date){
-        DateFormat monthDayYear = new SimpleDateFormat("dd/MM/yyyy");
-        String dateString = monthDayYear.format(date);
+        String dateString = formatter.format(date);
         return dateString;
     }
 
+    private static Date convertStringToDate(String dateString){
+        try{
+            Date date = formatter.parse(dateString);
+            return date;
+        } catch(ParseException e){
+            e.printStackTrace();
```

Comment:
prefer log details about error instead of printing the stack trace.

Check out the example in Java Logging Examples
the "simple use" will be helpfull. 
https://docs.oracle.com/javase/10/core/java-logging-overview.htm#JSCOR-GUID-48004124-2C00-49F7-A640-0C0DDA271DBC